### PR TITLE
fix[CL]: create position with underlying denoms only

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

We currently don't check the denoms when amounts are provided for the createPosition logic. 

Additionally, we take a single coin object for both amount0 and amount1. This becomes problematic when wanting to provide liquidity in a range that only requires a single asset, since a zero coin amount make a coin invalid.

This PR fixes the above two issues by accepting a single coins object and checks the underlying denoms before running the createPosition logic.

## Brief Changelog

- Checks amounts of pool token0 and pool token1 in the provided coins for createPosition
- Removed tokenDesired0 and tokenDesired1 from MsgCreatePosition in favor of tokensProvided
- Updates all tests that use createPosition to utilize Coins object instead of two individual Coin objects
- Requires CreateFullRangePosition to have a coins length of exactly 2
- Exapands validateBasic for MsgCreatePosition

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by mattverse** [comment]
rip merge conflicts with the msg create position PR 😭

File: x/concentrated-liquidity/lp_test.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp_test.go:55]: This work around was the cleanest way to allow a nil sdk.Coins{} value to be provided in the test case and not get overridden.

**Comment by czarcas7ic** [comment]
@mattverse we can get yours in first and I will correct the merge conflicts

File: proto/osmosis/concentrated-liquidity/tx.proto
Comment (p0mvn) [inline proto/osmosis/concentrated-liquidity/tx.proto:32]: Why do we have these as coins? Why can't we assume that if a user attempts to create a position in a pool where pool has denom A as token 0 and B as token 1, then the denoms are implied?

File: x/concentrated-liquidity/types/msgs.go
Comment (p0mvn) [inline x/concentrated-liquidity/types/msgs.go:45]: Since there are always 2 tokens expected, I think it is better UX to have 2 separate fields - one for token0 and another one for token1.

This way, people would not be able to shoot themselves in the foot by providing more tokens and failing the check here

Comment (czarcas7ic) [reply x/concentrated-liquidity/types/msgs.go:45]: (this is what you were suggesting, I am just pointing out that just because the two fields make sense here does not mean we should also have it for createPosition as well)

File: proto/osmosis/concentrated-liquidity/tx.proto
Comment (p0mvn) [reply proto/osmosis/concentrated-liquidity/tx.proto:32]: discussed offline, sg

File: x/concentrated-liquidity/types/msgs.go
Comment (p0mvn) [reply x/concentrated-liquidity/types/msgs.go:45]: Ah it's because we cannot have a zero coin

**Review by p0mvn — APPROVED** [review]
LGTM

**Review by mattverse — APPROVED** [review]
Leaving some questions I would like to get an answer for before merge, otherwise lgtm

File: proto/osmosis/concentrated-liquidity/tx.proto
Comment (czarcas7ic) [reply proto/osmosis/concentrated-liquidity/tx.proto:34]: proto comment added https://github.com/osmosis-labs/osmosis/pull/5119/commits/7d47173e47f7320679d89f75b45fa45b4b6976fd

File: x/concentrated-liquidity/lp.go
Comment (czarcas7ic) [reply x/concentrated-liquidity/lp.go:47]: No, since if both are zero then the liquidityDelta that is calculated is zero. That being said, I think you are right if your point was we should catch this earlier, will add this now

Comment (czarcas7ic) [reply x/concentrated-liquidity/lp.go:47]: https://github.com/osmosis-labs/osmosis/pull/5119/commits/cbdc89814db9c94515e42522554b43346cdec3aa

**Pull request merged**
